src/mxbase.jl

```diff
@@ -1,104 +1,72 @@
-libmx = C_NULL
-libeng = C_NULL
-libmat = C_NULL
-
 # Determine MATLAB library path and provide facilities to load libraries with
 # this path
 
-function get_paths()
-    global matlab_homepath = get(ENV, "MATLAB_HOME", "")
-    global default_startcmd = C_NULL
-    global matlab_library_path = nothing
-
-    if matlab_homepath == ""
+function matlab_home_path()
+    matlab_home = get(ENV, "MATLAB_HOME", "")
+    if matlab_home == ""
         if is_linux()
-            matlab_homepath = dirname(dirname(realpath(chomp(readstring(`which matlab`)))))
+            matlab_home = dirname(dirname(realpath(chomp(readstring(`which matlab`)))))
         elseif is_apple()
-            apps = readdir("/Applications")
-            filter!(app -> ismatch(r"^MATLAB_R[0-9]+[ab]\.app$", app), apps)
-            if ~isempty(apps)
-                matlab_homepath = joinpath("/Applications", maximum(apps))
+            default_dir = "/Applications"
+            if isdir(default_dir)
+                dirs = readdir(default_dir)
+                filter!(app -> ismatch(r"^MATLAB_R[0-9]+[ab]\.app$", dirs), dirs)
             end
         elseif is_windows()
             default_dir = Sys.WORD_SIZE == 32 ? "C:\\Program Files (x86)\\MATLAB" : "C:\\Program Files\\MATLAB"
             if isdir(default_dir)
                 dirs = readdir(default_dir)
                 filter!(dir -> ismatch(r"^R[0-9]+[ab]$", dir), dirs)
-                if ~isempty(dirs)
-                    matlab_homepath = joinpath(default_dir, maximum(dirs))
-                end
             end
         end
     end
+    if ~isempty(dirs)
```

Comment:
I actually mean that the dirs check needs to be in the if above since it is not defined if the branch is not executed